mavros_extras/src/plugins/gimbal_control.cpp

```diff
@@ -0,0 +1,708 @@
+      transform.transform.rotation = gimbal_attitude_msg.q;
+      uas->tf2_broadcaster.sendTransform(transform);
+    }
+  }
+
+  // Mavlink subscriber callbacks
+  /**
+   * @brief Publish the gimbal orientation
+   *
+   * Message specification: https://mavlink.io/en/messages/common.html#GIMBAL_DEVICE_ATTITUDE_STATUS
+   * @param msg - the mavlink message
+   * @param mo - received GimbalDeviceAttitudeStatus msg
+   */
+  void handle_gimbal_attitude_status(
+    const mavlink::mavlink_message_t * msg [[maybe_unused]],
+    mavlink::common::msg::GIMBAL_DEVICE_ATTITUDE_STATUS & mo,
+    plugin::filter::SystemAndOk filter [[maybe_unused]])
+  {
+    mavros_msgs::msg::GimbalDeviceAttitudeStatus gimbal_attitude_msg;
+    gimbal_attitude_msg.header = uas->synchronized_header(frame_id, mo.time_boot_ms);
+    gimbal_attitude_msg.target_system = mo.target_system;
+    gimbal_attitude_msg.target_component = mo.target_component;
+    gimbal_attitude_msg.flags = mo.flags;
+    geometry_msgs::msg::Quaternion q;
+    q.w = mo.q[0];
+    q.x = mo.q[1];
+    q.y = mo.q[2];
+    q.z = mo.q[3];
+    gimbal_attitude_msg.q = q;
+    gimbal_attitude_msg.angular_velocity_x = mo.angular_velocity_x;
+    gimbal_attitude_msg.angular_velocity_y = mo.angular_velocity_y;
+    gimbal_attitude_msg.angular_velocity_z = mo.angular_velocity_z;
+    gimbal_attitude_msg.failure_flags = mo.failure_flags;
+
+    gimbal_attitude_status_pub->publish(gimbal_attitude_msg);
+
+    // publish tf
+    publish_tf(gimbal_attitude_msg);
+  }
+
+  /**
+   * @brief Publish gimbal manager status
+   *
+   * Message specification: https://mavlink.io/en/messages/common.html#GIMBAL_MANAGER_STATUS
+   * @param msg - the mavlink message
+   * @param ms - received ManagerStatus msg
+   */
+  void handle_manager_status(
+    const mavlink::mavlink_message_t * msg [[maybe_unused]],
+    mavlink::common::msg::GIMBAL_MANAGER_STATUS & ms,
+    plugin::filter::SystemAndOk filter [[maybe_unused]])
+  {
+    mavros_msgs::msg::GimbalManagerStatus gimbal_manager_status_msg;
+    gimbal_manager_status_msg.header = uas->synchronized_header(frame_id, ms.time_boot_ms);
+    gimbal_manager_status_msg.flags = ms.flags;
+    gimbal_manager_status_msg.gimbal_device_id = ms.gimbal_device_id;
+    gimbal_manager_status_msg.sysid_primary = ms.primary_control_sysid;
+    gimbal_manager_status_msg.compid_primary = ms.primary_control_compid;
+    gimbal_manager_status_msg.sysid_secondary = ms.secondary_control_sysid;
+    gimbal_manager_status_msg.compid_secondary = ms.secondary_control_compid;
+
+    gimbal_manager_status_pub->publish(gimbal_manager_status_msg);
+  }
+
+  /**
+   * @brief Helper function for converting char arrays from GimbalDeviceInformation msg to string
+   * 
+   * @param a - array of chars from GimbalDeviceInformation Mavlink msg
+   * @param size - The size of the char array
+   */
+  std::string convertToString(std::array<char, 32> a, int size) {
```

Comment:
https://github.com/ArduPilot/pymavlink/blob/341d8892d443979bb467ff2619e82c61495fc3d5/generator/CPP11/include_v2.0/message.hpp#L130-L134

Reply:
Actually no, that one:

https://github.com/ArduPilot/pymavlink/blob/341d8892d443979bb467ff2619e82c61495fc3d5/generator/CPP11/include_v2.0/message.hpp#L89-L98